frontend/src/libs/components/form/form.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback } from 'react';
+import { type DeepPartial, type FieldValues } from 'react-hook-form';
+import { type ValidationSchema } from 'shared/build/index.js';
+
+import { useAppForm } from '~/libs/hooks/hooks.js';
+import { type FormField } from '~/libs/types/form.type.js';
+
+import { Button } from '../components.js';
```

Comment:
it could be circular dependency when you will add Form component to libs/components/components file. 
To avoid circular dep, rewrite this import directly to button, as you are already in components folder --> as you did it with input